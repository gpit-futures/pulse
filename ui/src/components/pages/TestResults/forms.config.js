export const valuesNames = {
  NAME: 'testName',
  STATUS: 'status',
  TAKEN: 'sampleTaken',
  CONCLUSION: 'conclusion',
  AUTHOR: 'author',
  DATE: 'dateCreated',

  TR: 'testResults',
  TR_RESULT: 'result',
  TR_VALUE: 'value',
  TR_UNIT: 'unit',
  TR_NORMAL: 'normalRange',
  TR_COMMENT: 'comment',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  NAME: 'Test Name',
  STATUS: 'Status',
  TAKEN: 'Sample Taken',
  CONCLUSION: 'Conclusion',
  AUTHOR: 'Author',
  DATE: 'Date',

  TR: 'Test Results',
  TR_RESULT: 'Results',
  TR_VALUE: 'Value',
  TR_UNIT: 'Unit',
  TR_NORMAL: 'Normal Range',
  TR_COMMENT: 'Comment',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
