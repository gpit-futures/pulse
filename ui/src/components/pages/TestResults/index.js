import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientTestResultsDetailEpic } from './ducks/fetch-patient-test-results-detail.duck';
import { fetchPatientTestResultsEpic } from './ducks/fetch-patient-test-results.duck';

import patientsTestResults from './ducks/fetch-patient-test-results.duck';
import testResultsDetail from './ducks/fetch-patient-test-results-detail.duck';

const epics = combineEpics(fetchPatientTestResultsDetailEpic, fetchPatientTestResultsEpic);
const TestResults = asyncComponent(() => import(/* webpackChunkName: "results" */ './TestResults').then(module => module.default));

const reducers = {
  patientsTestResults,
  testResultsDetail
};

const sidebarConfig = { key: 'results', pathToTransition: '/results', name: 'Test Results', isVisible: true };

const routers = [
  { key: 'testResults', component: TestResults, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.TEST_RESULTS}` },
  { key: 'testResultsDetail', component: TestResults, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.TEST_RESULTS}/:sourceId` },
];

export default {
  component: TestResults,
  epics, reducers, sidebarConfig, routers,
}

