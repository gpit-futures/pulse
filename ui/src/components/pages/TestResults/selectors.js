import { createSelector } from 'reselect';
import _ from 'lodash/fp';

import { operationsOnCollection } from '../../../utils/plugin-helpers.utils';
import { valuesNames } from './forms.config';

const patientTestResultsSelector = createSelector(
  ({ patientsTestResults }) => patientsTestResults,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsTestResults, userId) => {
    const allTestResults = operationsOnCollection.modificate(patientsTestResults[userId], [{
      key: valuesNames.TAKEN,
      fn: item => new Date(item).getTime(),
    }, {
      key: valuesNames.DATE,
      fn: item => new Date(item).getTime(),
    }]);
    return ({ allTestResults, userId });
  }
);

const patientTestResultsDetailSelector = createSelector(
  ({ testResultsDetail }) => testResultsDetail,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (testResultsDetail, userId) => {
    const testResultDetail = testResultsDetail[userId];
    return ({ testResultDetail, userId });
  }
);

export { patientTestResultsSelector, patientTestResultsDetailSelector }
