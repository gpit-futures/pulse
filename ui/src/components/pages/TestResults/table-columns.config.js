import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';


export const columnsConfig = [
  { key: 'testName', title: 'Test Name', width: '30%' },
  { key: 'sampleTaken', title: 'Sample Taken', transformer: getDDMMMYYYY, width: '26%' },
  { key: 'dateCreated', title: 'Date Created', transformer: getDDMMMYYYY, width: '25%' },
  { key: 'source', title: 'Source', width: '19%' },
  { key: 'sourceId', title: 'SourceID', display: 'none' },
];

export const defaultColumnsSelected = {
  testName: true,
  sampleTaken: true,
  dateCreated: true,
  source: true,
  sourceId: true,
};
