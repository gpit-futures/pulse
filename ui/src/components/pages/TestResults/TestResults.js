import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import classNames from 'classnames';
import _ from 'lodash/fp';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { lifecycle, compose } from 'recompose';

import PluginListHeader from '../../plugin-page-component/PluginListHeader';
import PluginMainPanel from '../../plugin-page-component/PluginMainPanel';

import { columnsConfig, defaultColumnsSelected } from './table-columns.config'
import { valuesNames } from './forms.config';
import { fetchPatientTestResultsRequest } from './ducks/fetch-patient-test-results.duck';
import { fetchPatientTestResultsDetailRequest } from './ducks/fetch-patient-test-results-detail.duck';
import { fetchPatientTestResultsOnMount, fetchPatientTestResultsDetailOnMount } from '../../../utils/HOCs/fetch-patients.utils';
import { patientTestResultsSelector, patientTestResultsDetailSelector } from './selectors';
import { clientUrls } from '../../../config/client-urls.constants';
import { operationsOnCollection } from '../../../utils/plugin-helpers.utils';
import TestResultsDetail from './TestResultsDetail/TestResultsDetail';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';

const TEST_RESULTS_MAIN = 'testResultsMain';
const TEST_RESULTS_DETAIL = 'testResultsDetail';
const TEST_RESULTS_CREATE = 'testResultsCreate';
const TEST_RESULT_PANEL = 'testResultPanel';
const META_PANEL = 'metaPanel';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ fetchPatientTestResultsRequest, fetchPatientTestResultsDetailRequest }, dispatch) });

@connect(patientTestResultsSelector, mapDispatchToProps)
@connect(patientTestResultsDetailSelector, mapDispatchToProps)
@compose(lifecycle(fetchPatientTestResultsOnMount), lifecycle(fetchPatientTestResultsDetailOnMount))
export default class TestResults extends PureComponent {
  static propTypes = {
    allTestResults: PropTypes.arrayOf(PropTypes.object),
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  state = {
    nameShouldInclude: '',
    selectedColumns: defaultColumnsSelected,
    openedPanel: TEST_RESULT_PANEL,
    columnNameSortBy: valuesNames.NAME,
    sortingOrder: 'asc',
    expandedPanel: 'all',
    isBtnExpandVisible: false,
    isAllPanelsVisible: false,
    isDetailPanelVisible: false,
    isSecondPanel: false,
    isCreatePanelVisible: false,
    editedPanel: {},
    offset: 0,
    isSubmit: false,
    isLoading: true,
  };

  componentWillReceiveProps() {
    const sourceId = this.context.router.route.match.params.sourceId;
    const userId = this.context.router.route.match.params.userId;

    //TODO should be implemented common function, and the state stored in the store Redux
    if (this.context.router.history.location.pathname === `${clientUrls.PATIENTS}/${userId}/${clientUrls.TEST_RESULTS}/${sourceId}` && sourceId !== undefined) {
      this.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false })
    }
    if (this.context.router.history.location.pathname === `${clientUrls.PATIENTS}/${userId}/${clientUrls.TEST_RESULTS}`) {
      this.setState({ isSecondPanel: false, isBtnExpandVisible: false, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: TEST_RESULT_PANEL, isDetailPanelVisible: false, expandedPanel: 'all' })
    }

    /* istanbul ignore next */
    setTimeout(() => {
      this.setState({ isLoading: false })
    }, 500)
  }

 handleExpand = (name, currentPanel) => {
   if (currentPanel === TEST_RESULTS_MAIN) {
     if (this.state.expandedPanel === 'all') {
       this.setState({ expandedPanel: name });
     } else {
       this.setState({ expandedPanel: 'all' });
     }
   } else if (this.state.expandedPanel === 'all') {
     this.setState({ expandedPanel: name, openedPanel: name });
   } else {
     this.setState({ expandedPanel: 'all' });
   }
 };

  handleFilterChange = ({ target: { value } }) => this.setState({ nameShouldInclude: _.toLower(value) });

  handleHeaderCellClick = (e, { name, sortingOrder }) => this.setState({ columnNameSortBy: name, sortingOrder });

  handleDetailTestResultsClick = (sourceId) => {
    const { actions, userId } = this.props;
    this.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, openedPanel: TEST_RESULT_PANEL, editedPanel: {}, expandedPanel: 'all', isLoading: true })
    actions.fetchPatientTestResultsDetailRequest({ userId, sourceId });
    this.context.router.history.push(`${clientUrls.PATIENTS}/${userId}/${clientUrls.TEST_RESULTS}/${sourceId}`);
  };

  handleSetOffset = offset => this.setState({ offset });

  handleCreate = () => {};

  handleEdit = (name) => {
    this.setState(prevState => ({
      editedPanel: {
        ...prevState.editedPanel,
        [name]: true,
      },
      isSubmit: false,
    }))
  };

  handleTestResultDetailCancel = (name) => {
    this.setState(prevState => ({
      editedPanel: {
        ...prevState.editedPanel,
        [name]: false,
      },
      isSubmit: false,
      isLoading: true,
    }))
  };

  handleSaveSettingsDetailForm = () => {};

  handleShow = (name) => {
    this.setState({ openedPanel: name })
  };

  formToShowCollection = (collection) => {
    const {columnNameSortBy, sortingOrder, nameShouldInclude} = this.state;

    collection = operationsOnCollection.modificate(collection, [{
      keyFrom: valuesNames.TAKEN,
      keyTo: `${valuesNames.TAKEN}Convert`,
      fn: getDDMMMYYYY
    }, {
      keyFrom: valuesNames.DATE,
      keyTo: `${valuesNames.DATE}Convert`,
      fn: getDDMMMYYYY
    }]);

    return operationsOnCollection.filterAndSort({
      collection: collection,
      filterBy: nameShouldInclude,
      sortingByKey: columnNameSortBy,
      sortingByOrder: sortingOrder,
      filterKeys: [valuesNames.NAME, `${valuesNames.TAKEN}Convert`, `${valuesNames.DATE}Convert`, valuesNames.SOURCE]
    });
  };

  render() {
    const { selectedColumns, columnNameSortBy, sortingOrder, isSecondPanel, isDetailPanelVisible, isBtnExpandVisible, expandedPanel, openedPanel, isBtnCreateVisible, isCreatePanelVisible, editedPanel, offset, isSubmit, isLoading } = this.state;
    const { allTestResults, testResultDetail } = this.props;

    const isPanelDetails = (expandedPanel === TEST_RESULTS_DETAIL || expandedPanel === TEST_RESULT_PANEL || expandedPanel === META_PANEL);
    const isPanelMain = (expandedPanel === TEST_RESULTS_MAIN);
    const isPanelCreate = (expandedPanel === TEST_RESULTS_CREATE);

    const columnsToShowConfig = columnsConfig.filter(columnConfig => selectedColumns[columnConfig.key]);

    const filteredTestResults = this.formToShowCollection(allTestResults);

    let sourceId;
    if (isDetailPanelVisible && !_.isEmpty(testResultDetail)) {
      sourceId = testResultDetail.sourceId;
    }

    return (<section className="page-wrapper">
      <div className={classNames('section', { 'full-panel full-panel-main': isPanelMain, 'full-panel full-panel-details': (isPanelDetails || isPanelCreate) })}>
        <Row>
          {(isPanelMain || expandedPanel === 'all')
            ? <Col xs={12} className={classNames({ 'col-panel-main': isSecondPanel })}>
              <div className="panel panel-primary">
                <PluginListHeader
                  onFilterChange={this.handleFilterChange}
                  panelTitle="Test Results"
                  isBtnExpandVisible={isBtnExpandVisible}
                  isBtnTableVisible={false}
                  name={TEST_RESULTS_MAIN}
                  onExpand={this.handleExpand}
                  currentPanel={TEST_RESULTS_MAIN}
                />
                <PluginMainPanel
                  headers={columnsToShowConfig}
                  resourceData={allTestResults}
                  emptyDataMessage="No test results"
                  onHeaderCellClick={this.handleHeaderCellClick}
                  onCellClick={this.handleDetailTestResultsClick}
                  columnNameSortBy={columnNameSortBy}
                  sortingOrder={sortingOrder}
                  table="testResults"
                  filteredData={filteredTestResults}
                  totalEntriesAmount={_.size(filteredTestResults)}
                  offset={offset}
                  setOffset={this.handleSetOffset}
                  onCreate={this.handleCreate}
                  id={sourceId}
                  isLoading={isLoading}
                />
              </div>
            </Col> : null}
          {(expandedPanel === 'all' || isPanelDetails) && isDetailPanelVisible && !isCreatePanelVisible
            ? <Col xs={12} className={classNames({ 'col-panel-details': isSecondPanel })}>
              <TestResultsDetail
                onExpand={this.handleExpand}
                name={TEST_RESULTS_DETAIL}
                openedPanel={openedPanel}
                onShow={this.handleShow}
                expandedPanel={expandedPanel}
                currentPanel={TEST_RESULTS_DETAIL}
                detail={testResultDetail}
                onEdit={this.handleEdit}
                editedPanel={editedPanel}
                onCancel={this.handleTestResultDetailCancel}
                onSaveSettings={this.handleSaveSettingsDetailForm}
                isSubmit={isSubmit}
              />
            </Col> : null}
        </Row>
      </div>
    </section>)
  }
}
