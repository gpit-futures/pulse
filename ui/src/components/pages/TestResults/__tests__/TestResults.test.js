import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import TestResults from '../TestResults';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsTestResults: {
    '9999999024': [
      {
        [valuesNames.DATE]: 1426997462000,
        currentDateConvert: '25-Dec-2017',
        [valuesNames.TAKEN]: 1424556662518,
        sampleTakenConvert: '22-Feb-2015',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: '1d27906b-ebc7-4848-9f1b-73187ee4f72e',
        [valuesNames.NAME]: 'Urea, electrolytes and creatinine measurement',
      }, {
        [valuesNames.DATE]: 1439953862000,
        currentDateConvert: '25-Dec-2017',
        [valuesNames.TAKEN]: 1440450662518,
        sampleTakenConvert: '25-Aug-2015',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: 'e54ffbfe-969e-4cae-bc5e-4850b298f5a4',
        [valuesNames.NAME]: 'complete blood count',
      },
    ],
  },
};

const storeEmpty = mockStore(Object.assign({
  testResultsDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  testResultsDetail: {
    '9999999000': {
      [valuesNames.AUTHOR]: 'Dr Lab',
      [valuesNames.CONCLUSION]: 'abnormal result indicating infection',
      [valuesNames.DATE]: 1439935862518,
      [valuesNames.TAKEN]: 1440454262518,
      [valuesNames.SOURCE]: 'marand',
      [valuesNames.SOURCE_ID]: 'f6763678-71dc-49c8-b6d1-88a5871a10a6',
      [valuesNames.STATUS]: 'Final',
      [valuesNames.NAME]: 'complete blood count',
      [valuesNames.TR]: [{
        [valuesNames.TR_COMMENT]: 'indicates infection',
        [valuesNames.TR_NORMAL]: ' - ',
        [valuesNames.TR_RESULT]: 'white blood cell count',
        [valuesNames.TR_UNIT]: '10*9/l',
        [valuesNames.TR_VALUE]: '13.6',
      }],
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/results`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextDetail = generateNewContext(context, `/patients/${userId}/results/${sourceId}`);

const match = {
  params: {
    userId,
  },
};

describe('Component <TestResults />', () => {
  it('should renders correctly with testResultPanel and testing Detail Panel', () => {
    const component = shallow(
      <TestResults
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive();

    // Testing component handleDetailTestResultsClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('TestResultsDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailTestResultsClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isCreatePanelVisible: false, openedPanel: 'testResultPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('TestResultsDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('testResultPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('testResultPanel', 'testResultsDetail');
    expect(component.state().openedPanel).toEqual('testResultPanel');
    expect(component.state().expandedPanel).toEqual('testResultPanel');
    component.setState({ openedPanel: 'testResultPanel', expandedPanel: 'testResultPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);

    component.instance().handleExpand('testResultsDetail', 'testResultsMain');
    expect(component.state().openedPanel).toEqual('testResultPanel');
    expect(component.state().expandedPanel).toEqual('all');

    component.setState({ expandedPanel: 'metaPanel' })
    component.instance().handleExpand('testResultsDetail', 'metaPanel');
    expect(component.state().expandedPanel).toEqual('all');

    component.instance().handleExpand('testResultsDetail', 'testResultsMain');
    expect(component.state().expandedPanel).toEqual('testResultsDetail');

    expect(component).toMatchSnapshot();

    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});
    component.instance().handleEdit('testResultPanel');
    expect(component.state().editedPanel).toEqual({ testResultPanel: true });
    component.instance().handleTestResultDetailCancel('testResultPanel');
    expect(component.state().editedPanel).toEqual({ testResultPanel: false });
    component.instance().handleEdit('testResultPanel');
    component.instance().handleSaveSettingsDetailForm();
    component.instance().handleCreate();

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <TestResults
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive();

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual(valuesNames.NAME);
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'author', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('author');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    // Testing component handleShow methods
    expect(component.state().openedPanel).toEqual('testResultPanel');
    component.instance().handleShow('metaPanel');
    expect(component.state().openedPanel).toEqual('metaPanel');


    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <TestResults
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <TestResults
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive()
      .dive();

    expect(component).toMatchSnapshot();
  });
});
