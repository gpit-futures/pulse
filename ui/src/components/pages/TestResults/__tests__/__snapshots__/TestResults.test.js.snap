// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TestResults /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="testResultsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="testResultsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Test Results"
          />
          <PluginMainPanel
            columnNameSortBy="testName"
            emptyDataMessage="No test results"
            headers={
              Array [
                Object {
                  "key": "testName",
                  "title": "Test Name",
                  "width": "30%",
                },
                Object {
                  "key": "sampleTaken",
                  "title": "Sample Taken",
                  "transformer": [Function],
                  "width": "26%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="testResults"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <TestResults /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="testResultsMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="testResultsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Test Results"
          />
          <PluginMainPanel
            columnNameSortBy="testName"
            emptyDataMessage="No test results"
            headers={
              Array [
                Object {
                  "key": "testName",
                  "title": "Test Name",
                  "width": "30%",
                },
                Object {
                  "key": "sampleTaken",
                  "title": "Sample Taken",
                  "transformer": [Function],
                  "width": "26%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            id="f6763678-71dc-49c8-b6d1-88a5871a10a6"
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="testResults"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <TestResultsDetail
          currentPanel="testResultsDetail"
          detail={
            Object {
              "author": "Dr Lab",
              "conclusion": "abnormal result indicating infection",
              "dateCreated": 1439935862518,
              "sampleTaken": 1440454262518,
              "source": "marand",
              "sourceId": "f6763678-71dc-49c8-b6d1-88a5871a10a6",
              "status": "Final",
              "testName": "complete blood count",
              "testResults": Array [
                Object {
                  "comment": "indicates infection",
                  "normalRange": " - ",
                  "result": "white blood cell count",
                  "unit": "10*9/l",
                  "value": "13.6",
                },
              ],
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="testResultsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="testResultPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <TestResults /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="testResultsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="testResultsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Test Results"
          />
          <PluginMainPanel
            columnNameSortBy="testName"
            emptyDataMessage="No test results"
            headers={
              Array [
                Object {
                  "key": "testName",
                  "title": "Test Name",
                  "width": "30%",
                },
                Object {
                  "key": "sampleTaken",
                  "title": "Sample Taken",
                  "transformer": [Function],
                  "width": "26%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="testResults"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <TestResults /> should renders correctly with testResultPanel and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="testResultsMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="testResultsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Test Results"
          />
          <PluginMainPanel
            columnNameSortBy="testName"
            emptyDataMessage="No test results"
            headers={
              Array [
                Object {
                  "key": "testName",
                  "title": "Test Name",
                  "width": "30%",
                },
                Object {
                  "key": "sampleTaken",
                  "title": "Sample Taken",
                  "transformer": [Function],
                  "width": "26%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            id="f6763678-71dc-49c8-b6d1-88a5871a10a6"
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="testResults"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <TestResultsDetail
          currentPanel="testResultsDetail"
          detail={
            Object {
              "author": "Dr Lab",
              "conclusion": "abnormal result indicating infection",
              "dateCreated": 1439935862518,
              "sampleTaken": 1440454262518,
              "source": "marand",
              "sourceId": "f6763678-71dc-49c8-b6d1-88a5871a10a6",
              "status": "Final",
              "testName": "complete blood count",
              "testResults": Array [
                Object {
                  "comment": "indicates infection",
                  "normalRange": " - ",
                  "result": "white blood cell count",
                  "unit": "10*9/l",
                  "value": "13.6",
                },
              ],
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="testResultsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="testResultPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <TestResults /> should renders correctly with testResultPanel and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <TestResultsDetail
          currentPanel="testResultsDetail"
          detail={
            Object {
              "author": "Dr Lab",
              "conclusion": "abnormal result indicating infection",
              "dateCreated": 1439935862518,
              "sampleTaken": 1440454262518,
              "source": "marand",
              "sourceId": "f6763678-71dc-49c8-b6d1-88a5871a10a6",
              "status": "Final",
              "testName": "complete blood count",
              "testResults": Array [
                Object {
                  "comment": "indicates infection",
                  "normalRange": " - ",
                  "result": "white blood cell count",
                  "unit": "10*9/l",
                  "value": "13.6",
                },
              ],
            }
          }
          editedPanel={Object {}}
          expandedPanel="metaPanel"
          isSubmit={false}
          name="testResultsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="testResultPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <TestResults /> should renders correctly with testResultPanel and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <TestResultsDetail
          currentPanel="testResultsDetail"
          detail={
            Object {
              "author": "Dr Lab",
              "conclusion": "abnormal result indicating infection",
              "dateCreated": 1439935862518,
              "sampleTaken": 1440454262518,
              "source": "marand",
              "sourceId": "f6763678-71dc-49c8-b6d1-88a5871a10a6",
              "status": "Final",
              "testName": "complete blood count",
              "testResults": Array [
                Object {
                  "comment": "indicates infection",
                  "normalRange": " - ",
                  "result": "white blood cell count",
                  "unit": "10*9/l",
                  "value": "13.6",
                },
              ],
            }
          }
          editedPanel={Object {}}
          expandedPanel="metaPanel"
          isSubmit={false}
          name="testResultsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="testResultPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;
